src/main/java/conversion/BaseServlet.java

```diff
@@ -188,6 +193,23 @@ private String getFileName(final Part part) {
         }
         return null;
     }
+    
+    /**
+     * Gets the full URL before the part containing the path(s) specified in urlPatterns of the servlet.
+     * @param request
+     * @return protocol://servername/contextPath 
+     */
+    protected static String getContextURL(final HttpServletRequest request) {        
+        StringBuffer full = request.getRequestURL();
+        String context = request.getContextPath();
+        int index = full.indexOf(context);        
+        if (index == -1) {
+            // Better to return something rather than nothing? Or should exception be thrown?
```

Comment:
indexOf() returns -1 if the given substring does not appear in the string. Not sure of a condition where the contextPath would be empty/different but I thought it would be better to cover that base.

Reply:
I don't think this should ever happen, so we can probably remove this check.

Reply:
The value returned here is external facing (it's written out via the API). Therefore, any downstream code that relies on the this value would behave unexpectedly if we return "", so for that reason I don't think we should do that.

I would suggest looking at the Javadoc for getRequestUrl and getContextPath to see if this is an condition we actually need to handle.

Reply:
I've had a check over the javadoc and it mentions nothing about potential failure cases so I think we are good to remove the check in that respect. There is one occasion when the contextPath is "" (for the root context), which in my code would substring to "" (since indexOf("") == 0). So it would be worth adding a check in for a zero length string and acting accordingly.

Reply:
I have removed the -1 check and added a check for empty context (if the user has the buildvu at the root path /).